kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    // On success the `result` will contain the digest. On error it will contain
+    // the HMAC error value.
+    fn hash_done(&'a self, result: Result<&'a mut [u8], u32>);
+}
+
+pub const HMAC_KEY_SIZE: usize = 16;
+
+pub trait HMAC<'a> {
+    /// Enable the HMAC hardware.
+    /// Must be called before any other methods
+    fn init(&self);
```

Comment:
We generally want to avoid initialize methods in HILs #1035, #1112

Reply:
The symmetric encryption APIs have enable/disable functions, should those be reworked as well?